packages/api/cms-api/src/generator/generate-crud-input-validators.spec.ts

```diff
@@ -0,0 +1,305 @@
+                const source = parseSource(file.content);
+                const classes = source.getClasses();
+                const cls = classes[0];
+                const structure = cls.getStructure();
+
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                const prop = structure.properties![0];
+                expect(prop.name).toBe("dateAsString");
+                const lengthDecorator = prop.decorators?.find((decorator) => decorator.name === "Length");
+                expect(lengthDecorator).toBeDefined();
+                expect(lengthDecorator?.arguments).toEqual(["2", "5"]);
+
+                const importDeclarations = source.getImportDeclarations();
+                const lengthImport = importDeclarations.find((getImportDeclaration) =>
+                    getImportDeclaration.getNamedImports().some((namedImport) => namedImport.getName() === "Length"),
+                );
+                expect(lengthImport).toBeDefined();
+                expect(lengthImport?.getModuleSpecifierValue()).toBe("class-validator");
+
+                orm.close();
+            });
+        });
+
+        describe("relative path", () => {
+            it("should set IsValidRedirectSource decorator", async () => {
+                LazyMetadataStorage.load();
+                const orm = await MikroORM.init({
+                    type: "postgresql",
+                    dbName: "test-db",
+                    entities: [TestEntityWithRelativeImportDecorator],
+                });
+
+                const out = await generateCrud({ targetDirectory: __dirname }, orm.em.getMetadata().get("TestEntityWithRelativeImportDecorator"));
+                const lintedOut = await lintGeneratedFiles(out);
+                const file = lintedOut.find((file) => file.name === "dto/test-entity-with-relative-import-decorator.input.ts");
+                if (!file) throw new Error("File not found");
+                const source = parseSource(file.content);
+                const classes = source.getClasses();
+                const cls = classes[0];
+                const structure = cls.getStructure();
+
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                const prop = structure.properties![0];
+                expect(prop.name).toBe("source");
+                const decorators = prop.decorators?.map((i) => i.name);
+                expect(decorators).toContain("IsValidRedirectSource");
+
+                const importDeclarations = source.getImportDeclarations();
+                const isSlugImport = importDeclarations.find((getImportDeclaration) =>
+                    getImportDeclaration.getNamedImports().some((namedImport) => namedImport.getName() === "IsValidRedirectSource"),
+                );
+                expect(isSlugImport).toBeDefined();
+                expect(isSlugImport?.getModuleSpecifierValue()).toBe("../../redirects/validators/isValidRedirectSource");
```

Comment:
How would this work in a "normal" use case? Would the import be from `@comet/cms-api`?